lib/task_bunny/job_error.ex

```diff
@@ -55,6 +55,26 @@ defmodule TaskBunny.JobError do
     reject: false
   ]
 
+  @doc """
+  Take information related to the result and make some of them JSON encode safe.
+
+  Since raw body can be bigger as you retry, you do not want to put the information.
+  """
+  @spec get_result_info(t) :: map()
+  def get_result_info(job_error) do
+    Map.take(job_error, [
+      :error_type,
+      :exception,
+      :stacktrace,
+      :return_value,
+      :reason,
+      :failed_count,
+      :queue
+    ])
+    |> Enum.map(fn ({k, v}) -> {k, inspect(v)} end)
```

Comment:
Does this turn integers into strings?
And if so, is that desired behavior?

Reply:
Good spot. It's more for logging purpose. I thought it's a good idea to keep the logic simpler. But I can also do...

```elixir
Enum.map(fn 
  ({:failed_count, v}) -> {k, v}
  ({k, v}) -> {k, inspect(v)}
end)
```

Reply:
Mhmmm, I mean we can all read the string type.
It might be confusing though.

This is getting formatted with Poison, right?
Would it work to check for protocol?

`Poison.Encoder.impl_for(v)` and otherwise inspect?

Though that might be a hassle for nested.


Yeah just keeping it string is easiest.

Reply:
Yeah, that's exactly the journey I have gone through :)